### ISSUE TITLE
Duplicated scores when importing from stable

### ISSUE THREAD

**Issue by Givikap120** [ppy/osu#30269]
### Type

Game behaviour

### Bug description

How to reproduce (can't be reproduced manually):
1. Import scores from stable
2. Wait till scoreinfo version would be updated
3. Import scores from stable again

### Screenshots or videos

For scores (this one is not so old):
![image](https://github.com/user-attachments/assets/5051acf6-e2c9-4c55-850f-c08b532dc34c)

This one is even older import, so old that player id wasn't saved (they can't even be exported):
![image](https://github.com/user-attachments/assets/2cedade7-9940-481c-bd47-74ba62f1af09)

### Version

2024.1009.1

### Logs

Not needed

**Comment by ssz7-ch2** [comment]
duplicate of https://github.com/ppy/osu/issues/26588 for the beatmap part

**Comment by Givikap120** [comment]
> duplicate of #26588 for the beatmap part

idk, bdach told me that there were no issue on this regard
I guess I change it to "score-only"

**Comment by bdach** [comment]
> idk, bdach told me that there were no issue on this regard

contrary to what you might think, i do not have eidetic memory to recall every single open issue over the >1k issues open right now

**Comment by Knallli** [comment]
What is the status on this? If people want to switch over to the new client, something like this cannot be disregarded

**Comment by peppy** [comment]
> What is the status on this? If people want to switch over to the new client, something like this cannot be disregarded

It only happens on multiple imports, so it shouldn't stop people moving over to lazer in the standard scenario. We'll definitely look at fixing this though.

**Comment by peppy** [comment]
@Knallli are you able to provide an example `.osr` file from stable (or even your whole stable `replays` folder) to help understand what's going on here?

**Comment by Knallli** [comment]
@peppy 

So basically, Lazer started the workflow to import scores from its current classic scores.db. However, since the classic db got corrupted at some point/ got rebuilt with scores missing, I started a manual import in Lazer of my osu!/Data/r directory from stable install to get scores back. This imported the missing scores into Lazer, but also duplicated all already imported scores.

[r.zip](https://github.com/user-attachments/files/24501237/r.zip)

I have attached some of my scores from classic here. My whole replays (r) folder is too big with 1GB to attach here.

I guess this is also replicatable by just importing one score file twice in a row.

**Comment by peppy** [comment]
Except it's not? I arbitrarily imported one of your `.osr` files twice and it doesn't duplicate. Are you sure you're not hitting a super-edge-case scenario bsaed on importing from a database and then *manually* from underlying data...?

**Comment by Knallli** [comment]
Ah, maybe it was caused by running two file import workflows at once? Since I had like 12k scores to import, the ones from db and the ones by file, it ran in parallel.

Have you imported the scores I sent with drag & drop or by using the Button in Settings>Maintenance>General>Import Files? I used it with the import files button.

As a workaround since this is an edge-case, are scores hashed by any chance and is it possible to deduplicate them with a cli or similar? I cannot just delete all scores and reimport them from stable since I also have Lazer-only scored that would get wiped then.

I will try and upload my whole score folder

**Comment by Givikap120** [comment]
> Except it's not? I arbitrarily imported one of your `.osr` files twice and it doesn't duplicate. Are you sure you're not hitting a super-edge-case scenario bsaed on importing from a database and then _manually_ from underlying data...?

In current version it's impossible to replicate. The problem was when you was importing it in the old lazer version (where score format was different), and then importing it again in the new version.
It doesn't happens to me because I don't play on stable anyway, but there's 100s of duplicated scores in my database. I would probably code the script to run on the lazer db to purge them.
